src/graph/optimizer/rule/UnionAllIndexScanBaseRule.cpp

```diff
@@ -48,19 +48,18 @@ bool UnionAllIndexScanBaseRule::match(OptContext* ctx, const MatchedResult& matc
   auto conditionType = condition->kind();
 
   if (condition->isLogicalExpr()) {
-    // Case1: OR Expr
-    if (conditionType == ExprKind::kLogicalOr) {
-      return true;
-    }
-    // Case2: AND Expr
+    // Don't support XOR yet
+    if (conditionType == ExprKind::kLogicalXor) return false;
+    if (conditionType == ExprKind::kLogicalOr) return true;
     if (conditionType == ExprKind::kLogicalAnd &&
         graph::ExpressionUtils::findAny(static_cast<LogicalExpression*>(condition),
                                         {ExprKind::kRelIn})) {
       return true;
     }
     // Check logical operands
     for (auto operand : static_cast<const LogicalExpression*>(condition)->operands()) {
-      if (!operand->isRelExpr() || !operand->isLogicalExpr()) {
+      if (!operand->isRelExpr() || operand->kind() != Expression::Kind::kLogicalAnd ||
+          operand->kind() != Expression::Kind::kLogicalOr) {
```

Comment:
@Aiee  Plz help make sure  if `||` or '&&`  is needed here.

Reply:
Should keep it as it was originally. Why add another condition here?

Reply:
isLogicalExpr() contains LogicXor, but this rule doesn't expect logicXor.

Reply:
I don't understand the logic here. Which kinds of exprs are expected here?

Reply:
As the comments in the code states:
```
// The matched expression should be either a OR expression or an expression that could be
// rewrote to a OR expression. There are 3 scenarios.
//
// 1. OR expr. If OR expr has an IN expr operand that has a valid index, expand it to OR expr.
//
// 2. AND expr such as A in [a, b] AND B when A has a valid index, because it can be transformed to
// (A==a AND B) OR (A==b AND B)
//
// 3. IN expr with its list size > 1, such as A in [a, b] since it can be transformed to (A==a) OR
// (A==b).
```

Reply:
ok